pkg/endpoint/events.go

```diff
@@ -119,6 +120,61 @@ func (e *Endpoint) PolicyRevisionBumpEvent(rev uint64) {
 	}
 }
 
+/// EndpointNoTrackEvent contains all fields necessary to update the NOTRACK rules.
+type EndpointNoTrackEvent struct {
+	ep     *Endpoint
+	annoCB AnnotationsResolverCB
+}
+
+// Handle handles the NOTRACK rule update.
+func (ev *EndpointNoTrackEvent) Handle(res chan interface{}) {
+	var IP string
+
+	e := ev.ep
+	port, err := ev.annoCB(e.K8sNamespace, e.K8sPodName)
+	if err != nil {
+		res <- &EndpointRegenerationResult{
+			err: err,
+		}
+		return
+	}
+
+	if !option.Config.EnableIPv4 {
+		IP = e.IPv6.String()
+	} else {
+		IP = e.IPv4.String()
```

Comment:
I think we need to install rules for all the valid IPs [1] of an endpoint?

[1] https://github.com/cilium/cilium/blob/master/pkg/endpoint/endpoint.go#L2094

Reply:
Done.